### ISSUE TITLE
Model diverges if FP16 is enabled

### ISSUE THREAD

**Issue by gongbudaizhe** [microsoft/DeepSpeed#1356]
I am training a ViT Transformer using deepspeed with FP16 enabled. The loss decreases at the beginning, however it starts to increase once it reaches loss around 14. Below is an example (x-axis: iteration, y-axis: loss value):

![image](https://user-images.githubusercontent.com/5178646/132822449-97bdf60d-39be-472a-83da-33a88bf7590f.png)

**Comment by tjruwase** [comment]
@gongbudaizhe, thanks for reporting this problem. Can you please share more details, such as script and dataset, to help repro?

**Comment by Bournet** [comment]
Did you solve this problem, I also got the same problem @gongbudaizhe

**Comment by gongbudaizhe** [comment]
No, I have tried gradient clipping and smaller learning rate but the results are no good. More in-depth investigation is needed.

For now, I am using FairScale from facebook research which supports ZeRO with AMP and doesn't have the divergence problem. However, the training speed and memory consumption is inferior to DeepSpeed.

**Comment by tjruwase** [comment]
@gongbudaizhe or @Bournet, are you able to share details to help us repro the problem? Thanks.

**Comment by tjruwase** [comment]
@gongbudaizhe, just wanted to check if you could share a small repro to help us fix this issue? Thanks.

**Comment by tjruwase** [comment]
Closing for lack of response. Please reopen as appropriate.
